src/client/src/Client.c

```diff
@@ -20,6 +20,84 @@ extern UINT32 CLIENT_STATE_MACHINE_STATE_COUNT;
  */
 PKinesisVideoClient gKinesisVideoClient = NULL;
 
+
+/**
+ *
+ * @param timerId - timerId for timer
+ * @param currentTime - the current time when the call back was fired
+ * @param customData - pKinesisVideoClient, contains the streams and interval info
+ * @return
+ */
+STATUS checkIntermittentProducerCallback(UINT32 timerId, UINT64 currentTime, UINT64 customData)
+{
+    ENTERS();
+    UNUSED_PARAM(timerId);
+    STATUS retStatus = STATUS_SUCCESS;
+    PKinesisVideoClient pKinesisVideoClient = (PKinesisVideoClient) customData;
+    UINT32 i;
+    BOOL clientStreamsListLocked = FALSE, streamLocked = FALSE;
+    PKinesisVideoStream pCurrStream = NULL;
+    Frame eofr = EOFR_FRAME_INITIALIZER;
+
+    CHK(pKinesisVideoClient, STATUS_NULL_ARG);
+
+    pKinesisVideoClient->timerCallbackInvocationCount++;
+
+    // Lock the client streams list lock
+    pKinesisVideoClient->clientCallbacks.lockMutexFn(pKinesisVideoClient->clientCallbacks.customData,
+                                                     pKinesisVideoClient->base.streamListLock);
+    clientStreamsListLocked = TRUE;
+    for (i = 0; i < pKinesisVideoClient->deviceInfo.streamCount; i++) {
+        if (NULL != pKinesisVideoClient->streams[i]) {
+            pCurrStream = pKinesisVideoClient->streams[i];
+            // Validate stream isn't being shut down or is closed
+            if (!(pCurrStream->streamStopped || pCurrStream->streamClosed)) {
+                // Lock the Stream
+                pKinesisVideoClient->clientCallbacks.lockMutexFn(pKinesisVideoClient->clientCallbacks.customData,
+                                                                 pCurrStream->base.lock);
+                streamLocked = TRUE;
+                // Check if last PutFrame is older than max timeout, if so, send EoFR, if not, do nothing
+                // Ignoring currentTime it COULD be smaller than pCurrStream->lastPutFrametimestamp
+                // Due to this method entering but waiting on stream lock from putFrame call
+                if (IS_VALID_TIMESTAMP(pCurrStream->lastPutFrameTimestamp) &&
+                    (GETTIME() - pCurrStream->lastPutFrameTimestamp) > INTERMITTENT_PRODUCER_MAX_TIMEOUT) {
+                    CHK_STATUS(putKinesisVideoFrame(TO_STREAM_HANDLE(pCurrStream), &eofr));

```

Comment:
Perhaps you don't want to bail out here. Maybe if (STATUS_FAILED(retStatus = putFrame())) {DLOGW("Failed to submit auto eofr with 0x%08x", retStatus);}

Reply:
Oh yes good catch! If we have multiple streams this will bail out after the first one it hits that it fails on, if we have a bad stream that's always misbehaving then subsequent ones in the list may never get eofr, will make this change!